File: algorithms/leetcode/121-best-time-to-buy-and-sell-stock/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int profit = 0;
        
        for (int price : prices) {
            min = Math.min(price, min);
            profit = Math.max(profit, price - min);
        }
        
        return profit;
    }
}

File: leetcode/121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
const maxProfit = (prices) => {
    if (prices.length < 2) {
        return 0;
    }

    let min = prices[0];
    let profit = 0;

    for (let i = 1; i < prices.length; i++) {
        min = Math.min(prices[i], min);
        profit = Math.max(profit, prices[i] - min);
    }

    return profit;
};
